### ISSUE TITLE
Refactor hierarchical firewall policy variables

### ISSUE THREAD

**Issue by ludoo** [GoogleCloudPlatform/cloud-foundation-fabric#1570]
In the organization/folder modules:

- the `firewall_policies` variable mixes *policy* and *policy* rules
- the locals in the `firewall-policies.tf` file are very hard to parse

One example while naming mixes policies and rules

```hcl
_merged_rules = flatten([
    for policy, rules in local.firewall_policies : [
      for name, rule in rules : merge(rule, {
        policy = policy
        name   = name
      })
    ]
  ])
```

Also check description fields for all resource management modules.

**Comment by juliocc** [comment]
What do you think if we move hfw policies to its own module and we leave only the attachment in the resman modules?

**Comment by ludoo** [comment]
> What do you think if we move hfw policies to its own module and we leave only the attachment in the resman modules?

Hmmmm the factory is pretty convenient to have at org/folder, but I agree that it has very little relationship with resman. I both like it and prefer the simplicity of the current approach, so whatever you prefer works for me.

**Comment by ludoo** [comment]
In hindsight, if we leverage the existing net firewall policy to create the policy (and optionally use it for attachment too), and only leave attachments in the resource management modules, that is probably the right way to go.

+1 to your approach

**Comment by ludoo** [comment]
Had a look at the VPN Network Firewall Policy module, and the resource attributes for network policies/rules are a 1:1 match for hirerachical policies/rules.

What we could do is integrate hierarchical policies in the same module and backport the factory. Then replace the `project_id` variable with a `root_node` one which accepts a project, folder, or organization. The first type of node creates a network policy, the second and third one a hierarchical policy.

Attachments should be optionally managed in the module itself via an `attachments` variable, which takes VPC ids for network policy, org/folder ids for hierarchical policy. Attachments should still possible from individual VPCs/folders/org by passing in the policy id.

This is the variable set of the current module

![image](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/assets/66007/12131e1c-6499-40fc-900d-d5ac8fc6dc6e)

I agree with @juliocc that this would result in a much cleaner design. I'm going to draft this unless you all don't like the approach.

**Comment by juliocc** [comment]
sounds great, go for it!
